network/server.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2025 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"net/http"
+	"time"
+)
+
+func NewServer(addr string, h http.Handler) *http.Server {
+	var protocols http.Protocols
+	protocols.SetHTTP1(true)
+	protocols.SetHTTP2(true)
+	protocols.SetUnencryptedHTTP2(true)
+
+	return &http.Server{
+		Addr:      addr,
+		Handler:   h,
+		Protocols: &protocols,
+
+		// https://medium.com/a-journey-with-go/go-understand-and-mitigate-slowloris-attack-711c1b1403f6
+		ReadHeaderTimeout: time.Minute,
```

Comment:
Do we care about `MaxHeaderBytes` to make it configurable?

Reply:
Unsure - circling back on this PR I realize we can't just remove the h2c upgrade flow because that can break existing workloads.

In theory we need 

1. Add support for h2 prior knowledge
2. deprecated h2c upgrade flow
3. then eventually drop h2c upgrade flow in some future release